From 2dabd3a00a039617c9f5bbe05ec7cd3bbdb33a38 Mon Sep 17 00:00:00 2001
From: Yuantian Tang <andy.tang@nxp.com>
Date: Wed, 24 Apr 2019 09:19:12 +0800
Subject: [PATCH 2/6] clk: qoriq: Add ls1028a clock configuration

commit 95089f6a9e7acf97496e4c5321b5003a4d7295c1 upstream

Enable clock driver by adding clock configuration for ls1028a chip.

Signed-off-by: Yuantian Tang <andy.tang@nxp.com>
Signed-off-by: Stephen Boyd <sboyd@kernel.org>

Upstream-status: Backport from the upstream

Signed-off-by: Liu Haitao <haitao.liu@windriver.com>
---
 drivers/clk/clk-qoriq.c | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/drivers/clk/clk-qoriq.c b/drivers/clk/clk-qoriq.c
index ef055fb132bd..03e5f3ad977b 100644
--- a/drivers/clk/clk-qoriq.c
+++ b/drivers/clk/clk-qoriq.c
@@ -248,12 +248,11 @@ static const struct clockgen_muxinfo clockgen2_cmux_cgb = {
 
 static const struct clockgen_muxinfo ls1028a_hwa1 = {
 	{
-		{},
+		{ CLKSEL_VALID, PLATFORM_PLL, PLL_DIV1 },
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV1 },
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV2 },
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV3 },
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV4 },
-		/* clock domain Y is clock */
 		{},
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV2 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV3 },
@@ -262,31 +261,37 @@ static const struct clockgen_muxinfo ls1028a_hwa1 = {
 
 static const struct clockgen_muxinfo ls1028a_hwa2 = {
 	{
-		{},
+		{ CLKSEL_VALID, PLATFORM_PLL, PLL_DIV1 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV1 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV2 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV3 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV4 },
-		/* clock domain Y is clock */
 		{},
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV2 },
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV3 },
 	},
 };
 
-/* The definition hadn't be provided on reference manual currently */
 static const struct clockgen_muxinfo ls1028a_hwa3 = {
+	{
+		{ CLKSEL_VALID, PLATFORM_PLL, PLL_DIV1 },
+		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV1 },
+		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV2 },
+		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV3 },
+		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV4 },
+		{},
+		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV2 },
+		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV3 },
+	},
 };
 
 static const struct clockgen_muxinfo ls1028a_hwa4 = {
 	{
-		/* Sync mode operation wrt to clock domain X */
-		{},
+		{ CLKSEL_VALID, PLATFORM_PLL, PLL_DIV1 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV1 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV2 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV3 },
 		{ CLKSEL_VALID, CGA_PLL2, PLL_DIV4 },
-		/* clock domain Y is clock */
 		{},
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV2 },
 		{ CLKSEL_VALID, CGA_PLL1, PLL_DIV3 },
-- 
2.31.1

